aiida_fleur/data/fleurinp.py

```diff
@@ -416,7 +424,28 @@ def _set_inp_dict(self):
 
         # check if it validates against the schema
         if not xmlschema.validate(tree_x):
-            raise InputValidationError("Input file is not validated against the schema.")
+            # get a more information on what does not validate
+            message = ''
+            try:
+                parser = etree.XMLParser(
+                    attribute_defaults=True, schema=xmlschema
+                )  #dtd_validation=True,
+                inpxmlfile = self.open(key='inp.xml', mode='r')
+                try:
+                    tree_x = etree.parse(inpxmlfile, parser)
+                finally:
+                    inpxmlfile.close()
+                #all errors will be raised
+            except etree.XMLSyntaxError as msg:
+                message = msg
+                raise InputValidationError(
+                    "Input file does not validate against the schema: {}".format(message)
+                )
+
+            #other error occured, we still stop
+            raise InputValidationError(
+                "Input file does not validate against the schema, some other error occured, (no XMLSyntaxError).}"
+            )
```

Comment:
Earlier I removed schema validation at parsing time because XInclude parts could not be validated before substitution. I think this part of code will be suffer from the same problem